spec.md

```diff
@@ -0,0 +1,948 @@
+Since:        0.2.0
+```
+
+The `EDITION` indicates what edition of games a song belongs to.
+While this header is intended to hold commercially available editions (e.g. SingStar Pop Hits, GuitarHero Live) implementations MUST NOT reject a file based on the value of this header.
+A list of SingStar editions is available [here](https://github.com/bohning/usdb_syncer/wiki/SingStar-Editions).
+For arbitrary keywords see the `TAGS` header.
+
+### 3.23. The `TAGS` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        1.1.0
+```
+
+The `TAGS` allow association of any reasonable keyword with a song.
+Implementations SHOULD compare tags in a case-insensitive manner.
+
+> [!CAUTION]
+>
+> Whether tags should be compared case-insensitively or not hasn't been decided yet.
+
+### 3.24. The `P1`, `P2`, … Headers
+
+```
+Required:     No
+Multi-Valued: No
+Since:        0.2.0
+```
+
+The headers `P1`, `P2`, … indicate the names of the voices of a song.
+These names correspond to the voices indicated by the `P1`, `P2`, … voice changes (see [section 3.3](#33-voice-changes)).
+If the voices correspond to different singers in the original song, the header values often indicate the names of the original singers.
+
+The association of header values to voices is defined by the numerical value after each `P` respectively,
+i.e. the header `P2` indicates the name of the voice whose notes are introduced by the `P2` voice change.
+Leading zeroes are insignificant, i.e. the headers `P1` and `P001` both refer to the same voice.
+
+> [!CAUTION]
+>
+> The exact semantics of the `P` headers have not been decided yet.
+
+### 3.25. The `DUETSINGER1`, `DUETSINGER2`, … Headers
+
+```
+Required:     No
+Multi-Valued: No
+Since:        0.2.0
+Deprecated:   0.3.0
+Removed:      1.0.0
+```
+
+The headers `DUETSINGER1`, `DUETSINGER2`, etc. are aliases for `P1`, `P2`, etc.
+If both are specified `P1`, `P2`, etc. take precedence.
+
+### 3.26. The `CREATOR` Header
+
+```
+Required:     No
+Multi-Valued: Yes
+Since:        0.2.0
+```
+
+The `CREATOR` indicates who created the textfile.
+Values are usually usernames or gamer tags.
+
+> [!NOTE]
+>
+> Some implementations are known to use an application-specific header `AUTHOR` in place of `CREATOR`.
+> The semantics of the `AUTHOR` header are not part of this specification.
+
+### 3.27. The `PROVIDEDBY` Header
+
+```
+Required:     No
+Multi-Valued: No
+Since:        1.1.0
+```
+
+The `PROVIDEDBY` header indicates the source of a particular textfile.
+Implementations concerned with providing textfiles to many users (sometimes referred to as "hosters") SHOULD set this value automatically.
+Values SHOULD be valid URLs according to [RFC 1738](https://datatracker.ietf.org/doc/html/rfc1738) using the HTTP or HTTPS scheme.
+
+> [!NOTE]
+>
+> Some implementations are known to use an application-specific header `SOURCE` in place of `PROVIDEDBY`.
+> The semantics of the `SOURCE` header are not part of this specification.
```

Comment:
I think this note can be removed since it was mentioned in a PR for performous, but not within the game yet. Unless other applications use this too?

Reply:
Ah I didn't realize that. I removed the note.